Tests/MapboxNavigationTests/NavigationCameraTests.swift

```diff
@@ -0,0 +1,346 @@
+        XCTAssertNil(overviewCarPlayCameraOptions?.zoom, "Zoom should be nil.")
+        XCTAssertNil(overviewCarPlayCameraOptions?.pitch, "Pitch should be nil.")
+    }
+    
+    func testViewportDataSourceDelegateForActiveGuidance() {
+        let navigationMapView = NavigationMapView(frame: .zero)
+        
+        // Create new `NavigationViewportDataSource` instance, which listens to the
+        // `Notification.Name.routeControllerProgressDidChange` notification, which is sent during
+        // active guidance navigation.
+        let navigationViewportDataSource = NavigationViewportDataSource(navigationMapView.mapView,
+                                                                        viewportDataSourceType: .active)
+        
+        navigationMapView.navigationCamera.viewportDataSource = navigationViewportDataSource
+        
+        let viewportDataSourceDelegateMock = ViewportDataSourceDelegateMock()
+        navigationMapView.navigationCamera.viewportDataSource.delegate = viewportDataSourceDelegateMock
+        
+        let origin = CLLocationCoordinate2DMake(37.765469, -122.415279)
+        let destination = CLLocationCoordinate2DMake(37.767071968183814, -122.41340145370796)
+        let routeOptions = NavigationRouteOptions(coordinates: [origin, destination])
+        
+        // Load previously serialized `Route` object in JSON format and deserialize it.
+        let routeData = Fixture.JSONFromFileNamed(name: "route-for-navigation-camera")
+        let decoder = JSONDecoder()
+        decoder.userInfo[.options] = routeOptions
+        
+        guard let route = try? decoder.decode(Route.self, from: routeData) else {
+            XCTFail("Route should be valid.")
+            return
+        }
+        
+        let routeProgress = RouteProgress(route: route,
+                                          routeIndex: 0,
+                                          options: routeOptions)
+        
+        let location = CLLocation(latitude: 37.765469, longitude: -122.415279)
+        
+        // Since second `stepIndex` is right after sharp maneuver default navigation camera behavior
+        // will change `CameraOptions.pitch` to `FollowingCameraOptions.defaultPitch`.
+        routeProgress.currentLegProgress.stepIndex = 1
+        
+        let userInfo: [RouteController.NotificationUserInfoKey: Any] = [
+            .routeProgressKey: routeProgress,
+            .locationKey: location,
+        ]
+        
+        let expectation = self.expectation(forNotification: .routeControllerProgressDidChange,
+                                           object: self) { _ in
+            return true
+        }
+        
+        NotificationCenter.default.post(name: .routeControllerProgressDidChange,
+                                        object: self,
+                                        userInfo: userInfo)
+        
+        wait(for: [expectation], timeout: 1.0)
```

Comment:
For you information, you can wait for all expectations with `open func waitForExpectations(timeout: TimeInterval, handler: XCWaitCompletionHandler? = nil)`